src/main/java/xyz/thepathfinder/android/Cluster.java

```diff
@@ -294,48 +348,275 @@ public Transport getTransport(String name) {
 
     /**
      * Returns an immutable collection of this cluster's transports.
+     *
      * @return A collection of transports.
      */
     public Collection<Transport> getTransports() {
-        return Collections.unmodifiableCollection(this.transports.values());
+        return Collections.<Transport>unmodifiableCollection(this.transports.values());
     }
 
-    public Transport removeTransport(Transport transport) {
-        return this.transports.remove(transport.getPath());
+    /**
+     * Returns an immutable map of this cluster's transports.
+     *
+     * @return A map of transports.
+     */
+    public Map<String, Transport> getTransportsMap() {
+        return Collections.<String, Transport>unmodifiableMap(this.transports);
     }
 
+    /**
+     * Sets this cluster's transports.
+     *
+     * @param transports an iterable collection of transports
+     */
     private void setTransports(Iterable<Transport> transports) {
-        for(Transport transport: transports) {
+        for (Transport transport : transports) {
             this.transports.put(transport.getPath(), transport);
         }
     }
 
+    /**
+     * Sets this cluster's transports.
+     *
+     * @param transports a map of transports
+     */
+    private void setTransports(Map<String, Transport> transports) {
+        this.transports = transports;
+    }
+
+    /**
+     * Sets this cluster's routes.
+     *
+     * @param routes a list of routes for this cluster
+     */
     private void setRoutes(List<Route> routes) {
         this.routes = routes;
     }
 
     /**
      * Returns an immutable collection of this cluster's routes.
+     *
      * @return A collection of routes.
      */
     public List<Route> getRoutes() {
         return this.routes;
     }
 
+    /**
+     * {@inheritDoc}
+     */
     @Override
     protected String getModel() {
```

Comment:
Here and a few other places, you seem to interchange Model as either being a String or a Model object. Which is it?